src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkValueProvider.cs

```diff
@@ -207,17 +212,27 @@ private async Task<bool> IsWPFApplicationAsync(IProjectProperties defaultPropert
                 {
                     // Enabled
 
-                    // Create the Application.xaml if it doesn't exist. We don't care about the path, we just need it to be created.
-                    string? appXamlFilePath = await GetAppXamlRelativeFilePathAsync(create: true);
-                    if (appXamlFilePath is not null)
+                    //project accessor to remove the applicationdefinition property if re-enabled
+                    await _projectAccessor.OpenProjectXmlForWriteAsync(_project, project =>
                     {
-                        IEnumerable<IProjectItem> matchingItems = await _sourceItemsProvider.GetItemsAsync(NoneItemType, appXamlFilePath);
-                        IProjectItem? appXamlItem = matchingItems.FirstOrDefault();
-                        if (appXamlItem is not null)
+                        foreach (ProjectItemGroupElement itemGroup in project.ItemGroups.ToList())
                         {
-                            await appXamlItem.SetItemTypeAsync(ApplicationDefinitionItemType);
+                            foreach (ProjectItemElement item in itemGroup.Items.ToList())
+                            {
+                                if (string.Equals(item.ItemType, ApplicationDefinitionItemType, StringComparisons.ItemTypes))
+                                {
+                                    itemGroup.RemoveChild(item);
+                                }
+                                else if (string.Equals(item.ItemType, NoneItemType, StringComparisons.ItemTypes) && string.Equals(item.Include, ApplicationFileName, StringComparison.Ordinal))
```

Comment:
Does this have to be ordinal? I think we would accept `application.xaml`, for example. Perhaps use `StringComparers.Paths`.